src/otx/core/data/transform_libs/mmpretrain.py

```diff
@@ -43,19 +55,70 @@ def transform(self, results: dict) -> MulticlassClsDataEntity:
         pad_shape = data_samples.metainfo.get("pad_shape", img_shape)
         scale_factor = data_samples.metainfo.get("scale_factor", (1.0, 1.0))
 
-        labels = results["__otx__"].labels
+        return {
+            "image": image,
+            "img_shape": img_shape,
+            "ori_shape": ori_shape,
+            "pad_shape": pad_shape,
+            "scale_factor": scale_factor,
+            "labels": results["__otx__"].labels,
+        }
+
+    def _pack_multiclass_inputs(self, results: dict) -> MulticlassClsDataEntity:
+        """Pack multiclass classification inputs."""
+        packed_common_inputs = self._pack_common_inputs(results)
 
         data_entity = MulticlassClsDataEntity(
-            image=image,
+            image=packed_common_inputs["image"],
+            img_info=ImageInfo(
+                img_idx=0,
+                img_shape=packed_common_inputs["img_shape"],
+                ori_shape=packed_common_inputs["ori_shape"],
+                scale_factor=packed_common_inputs["scale_factor"],
+            ),
+            labels=packed_common_inputs["labels"],
+        )
+        data_entity.img_info.pad_shape = packed_common_inputs["pad_shape"]
+        return data_entity
+
+    def _pack_multilabel_inputs(self, results: dict) -> MultilabelClsDataEntity:
+        """Pack multilabel classification inputs.
+
+        NOTE,
+        Currently, this function is the same with multiclass case.
+        However, it should have different functionality if we consider the ignore_label.
+        That's the reason why I reamined same function.
```

Comment:
I'm still doubt that we should split it. If there is `ignore_label` information, should it be considered when creating `OTXDataEntity`? Why do we need to handle at `PackInputs` which is effective only for `mmx` transforms.

Reply:
After some consideration, I think we can integrate all parts here. BTW, I want to do that at the next PR since now I'm focusing on checking the backward compatibility, what do you think?

Reply:
LGTM